sourcegraph/sourcegraph
# web: add `opentelemetry` `ClientAttributesSpanProcessor`

## Context

This PR adds [a simple span processor](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#span-processor) that adds attributes useful for every span created by the client application.

```ts
export enum ClientAttributes {
    LocationHref = 'window.location.href',
    LocationPathname = 'window.location.pathname',
    LocationSearch = 'window.location.search',
    AppVersion = 'app.version',
    BrowserName = 'browser.name',
}
```

Closes https://github.com/sourcegraph/sourcegraph/issues/40562

## Test plan

1. `ENABLE_OPEN_TELEMETRY=true sg start web-standalone`
2. Load the page.
3. See span attributes in the console.

## App preview:

- [Web](https://sg-web-vb-client-attributes-span.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-qcwgdqxdyc.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
